module/VuFind/src/VuFind/Search/Factory/BlenderBackendFactory.php

```diff
@@ -0,0 +1,185 @@
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   David Maus <maus@hab.de>
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     http://vufind.org   Main Site
+ */
+namespace VuFind\Search\Factory;
+
+use Interop\Container\ContainerInterface;
+use Laminas\EventManager\EventManager;
+use Laminas\ServiceManager\Factory\FactoryInterface;
+use VuFind\Search\Solr\DeduplicationListener;
+use VuFindSearch\Backend\Blender\Backend;
+use VuFindSearch\Backend\Solr\Backend as SolrBackend;
+
+/**
+ * Factory for Blender backend.
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   David Maus <maus@hab.de>
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     http://vufind.org   Main Site
+ */
+class BlenderBackendFactory implements FactoryInterface
+{
+    /**
+     * Superior service manager.
+     *
+     * @var ContainerInterface
+     */
+    protected $serviceLocator;
+
+    /**
+     * VuFind configuration reader
+     *
+     * @var \VuFind\Config\PluginManager
+     */
+    protected $config;
+
+    /**
+     * Search configuration file identifier.
+     *
+     * @var string
+     */
+    protected $searchConfig;
+
+    /**
+     * Facet configuration file identifier.
+     *
+     * @var string
+     */
+    protected $facetConfig;
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        $this->searchConfig = 'Blender';
+        $this->facetConfig = 'Blender';
+    }
+
+    /**
+     * Create service
+     *
+     * @param ContainerInterface $sm      Service manager
+     * @param string             $name    Requested service name (unused)
+     * @param array              $options Extra options (unused)
+     *
+     * @return Backend
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function __invoke(ContainerInterface $sm, $name, array $options = null)
+    {
+        $this->serviceLocator = $sm;
+        $this->config = $sm->get(\VuFind\Config\PluginManager::class);
+        $yamlReader = $sm->get(\VuFind\Config\YamlReader::class);
+        $blenderConfig = $this->config->get($this->searchConfig);
+        if (!isset($blenderConfig['Primary']['backend'])) {
```

Comment:
Any particular reason you're using array rather than object syntax here? We usually use object syntax when we're working with Laminas\Config\Config objects.

Reply:
It was probably just some legacy stuff, now changed.